### ISSUE TITLE
Testing your code in real data

### ISSUE THREAD

**Issue by dbergeba** [mvgolub/FW-DTI-Beltrami#1]
Hi, 
 I tested your code in a large sample of subjects using a single shell acquisition (b=800). The code run as expected, however, the values encountered are unexpectedly low and with too much variance. I mean, in gray matter, average free water values ranged from 0.1 to 0.25 (other scripts return around 0.3), and in white matter, also a huge variance and low values were found. Are there any issues or important updates that I should be aware of? Attached there is an example of the image.

Thanks in advance for your work.

![Screenshot from 2021-05-28 11-32-13](https://user-images.githubusercontent.com/34525523/119963507-77305800-bfa8-11eb-8bb4-a22808a27f20.png)

**Comment by mvgolub** [comment]
Hi,
thank you for testing the code, by the look of your screenshot, I assume
that you are using the "hybrid initialization", am I correct? If yes, are
you choosing your own  handpicked "St" and "Sw"? Are these values equal to
those used by other scripts (assuming the other scripts use the same type
of initialization)? Setting the correct pair of values can be tricky.

You could also try tweaking the learning rate and number of iterations to
see if regularization helps, although single-shell methods are highly
dependent on the initialization and regularization does little to help.

Do the FA and MD look normal?

**Comment by dbergeba** [comment]
Yes, 
 I used the "hybrid initialization". To set St and Sw values I just kept the method in the original script reproduced below:
"S0 = np.mean(masked_single[..., gtab_single.b0s_mask], axis=-1) * mask
St = np.round(np.percentile(S0[WM], 95))
Sw = np.round(np.percentile(S0[CSF], 95))"

Would you suggest another method?

Thanks

**Comment by mvgolub** [comment]
Sorry for the delayed response, you could try using the 'MD' initialization, it does not require setting up St and Sw, although the literature suggests that the 'hybrid' init is better.  Are the other scripts that you mentioned, publicly available? I could try to cross check the update rules for the gradient descent or the initialization equations, although it could take some time because I am currently working on another project.

On a second note, I now realized that some supplementary code was added for the paper acceptance, but this repo hasn't been updated yet (I will update the repo as soon as I can). The added code did not change the core functioning of the algorithm, but added some supplementary results, including a quantitative table of medians and interquartile ranges in WM and GM masks: Table 2 of the paper mentioned in the `README.md` (Free water DTI estimates from single b-value data might seem plausible but must be interpreted with care). In this table, single shell FW-DTI presented low median values for FW in WM and GM:  0.088 and 0.159, respectively, these values were more or less consistent with a reference multi-shell implementation: 0.107 and 0.117 for WM and GM respectively. If you don't have access to the paper, I can email you the table or even the full paper.

**Comment by dbergeba** [comment]
Thansk Marc,
 Finally I used another script from the MITk, not publicly available
though.  It returned values around 0.3 in gray matter, and 0.2 in white
matter, which corresponds to what I expected.

Thanks

Missatge de Marc ***@***.***> del dia dc., 2 de juny 2021 a
les 0:10:
